app/models/real_sitemap.rb

```diff
@@ -0,0 +1,13 @@
+class RealSitemap < ActiveRecord::Base
+  include Fetchable
+
+  attr_readonly :url
+
+  belongs_to :searchgov_domain
+
+  validates :url, presence: true, uniqueness: true,
```

Comment:
As I mentioned in our chat:

"You probably don’t need these validations in the Sitemap class, because you’re getting the validations in `Fetchable` automatically:
  ```
validates :url, presence: true, uniqueness: true,
   length: {maximum: 2000}, format: {with: /\Ahttps?:\/\/\w+\.gov\/\w+\.(xml|gz|txt)/, message: 'invalid url'}

```
Fetchable validations:
 ```
validates_length_of :url, maximum: 2000
validates_presence_of :url
validates_url :url, allow_blank: true
```
I should also note that the `validates_url` method comes courtesy of the `validate_url` gem, which takes care of the URL format validation (and don’t be surprised if some unexpected URLs are marked as “valid” - URLs are notoriously tricky to validate: https://mathiasbynens.be/demo/url-regex)"

Reply:
@MothOnMars I do think that `validates_url` isn't going to suffice completely - we still need to validate that the URL ends in `\.(xml|gz|txt)`, right?

Reply:
Nope - I specifically told Ethan not to validate the extension, because I've seen XML sitemaps with other extensions, such as `.aspx`. I'm trusting our super admins (and eventually our admins) not to put junk into that table. If we see too much Garbage In, I may want to lock down the validations at that point.

Reply:
Oh, OK. I was context-free on that part of the conversation. I think you can see how I inferred the requirement from this validation though.

Reply:
Yeah, that's the validation I told him to remove.

Reply:
So I was finding when running certain specs in `sitemap_spec.rb` that a test that made sure the URL 'https://foo/bar' was not valid would not pass, because a Sitemap object would still end up being created despite that (I think) we don't want that URL to be considered valid.  (I suppose I'm misunderstanding what we want out of our sitemap model, though.)

And now that I'm looking through that spreadsheet it looks like my validation doesn't come close to accommodating all those, so it was a fool's errand on my part to try to write a validation in the first place.